frontend/templates/product-list/index.tsx

```diff
@@ -2,14 +2,21 @@ import { InstantSearchProvider } from '@components/common/InstantSearchProvider'
 import { DefaultLayout } from '@layouts/default';
 import { ProductListTemplateProps } from './hooks/useProductListTemplateProps';
 import { ProductListView } from './ProductListView';
+import { useRouter } from 'next/router';
+import { getRouteData } from '@helpers/path-helpers';
 
 const ProductListTemplate: NextPageWithLayout<ProductListTemplateProps> = ({
    appProps,
    productList,
 }) => {
+   const router = useRouter();
+   const { deviceHandle } = getRouteData(router.asPath);
    return (
       <InstantSearchProvider {...appProps.algolia!}>
-         <ProductListView productList={productList} />
+         <ProductListView
+            productList={productList}
+            deviceHandle={deviceHandle}
+         />
```

Comment:
Instead of reaching into the router everywhere we need the device/variant path component, it would be nice to have a hook like `useDevicePartsVariant` or something. Similar to what we with the item type hook [`useDevicePartsItemType`](https://github.com/iFixit/react-commerce/blob/main/frontend/templates/product-list/hooks/useDevicePartsItemType.ts).

We could probably pull the variant out of `useCurrentRefinements()['woksin']` instead of the router too.

Reply:
[I created a `useVariant()`](https://github.com/iFixit/react-commerce/blob/main/frontend/templates/product-list/MetaTags.tsx#L103-L109)